docker/oss-fuzz-builder/Dockerfile

```diff
@@ -21,6 +21,7 @@ FROM $parent_image
 
```

Comment:
high level question (not strictly about this PR):  how difficult would it be to integrate imported oss-fuzz benchmarks with the fuzzbench builder/runner docker images, instead of using these oss-fuzz ones?

Reply:
I definitely want to minimize or eliminate the differences between OSS-Fuzz and standard benchmarks. This PR eliminates some of them.
After this PR lands there will be three main differences between OSS-Fuzz and standard benchmarks:
1. OSS-Fuzz benchmarks need to run checkout_commit.py (and hence need this image).
2. OSS-Fuzz benchmarks inherit from gcr.io/oss-fuzz/base-images/base-builder instead of gcr.io/fuzzbench/base-builder
3. OSS-Fuzz benchmarks inherit from a particular version of gcr.io/oss-fuzz/base-images/base-builder instead of "latest" like standard benchmarks do.

1. is a relatively minor difference, we could reduce it further by running the script for standard benchmarks and making it a noop, or we could eliminate it by making the integration script edit the docker file to checkout the right commit (or putting this (minor) burden on users).

2. is the biggest of differences/complexity. It can be solved by making all benchmarks inherit from gcr.io/oss-fuzz/base-images/base-builder. Or it can be solved by making all benchmarks inherit from gcr.io/fuzzbench/base-builder. I think gcr.io/oss-fuzz/base-images/base-builder makes more sense.

3. is done to ensure that OSS-Fuzz benchmarks build and can continue to build forever. By using the version of gcr.io/oss-fuzz/base-images/base-builder that has built the desired benchmark commit, we know that the build wont suddenly start failing because the compiler changed (in OSS-Fuzz as a whole compiler rolls frequently break projects). It's worth keeping in mind that non of our standard benchmarks use a pinned clang and have broken because of this. In the long run this will probably reduce maintenance burden (though we probably don't want to have benchmarks using clang 11.0 in 10 years from now when everyone is using clang 20 for example).

For now, I think the PR can land since it doesn't really increase differences, it makes eliminating them easier.

Maybe we could discuss this during the weekly meeting?

Reply:
Thanks for the details, this is very useful! Sure, let's continue discussing later, I was asking independently of the PR.